SD_DSC.adoc

```diff
@@ -517,14 +515,28 @@ Note that the primary purpose of Step #3. is to demonstrate that appropriate sea
 
 Step #6 ensures that partial key fragments do not remain in non-volatile memory. If the evaluator finds a 32-or-greater-consecutive-bit fragment, then fail immediately. Otherwise, there is a chance that it is not within the context of a key (e.g., some random bits that happen to match). If this is the case the test should be repeated with a different key in Step #1. If a fragment is also found in this repeated run then the test fails unless the developer provides a reasonable explanation for the collision, then the evaluator may give a pass on this test.
 
-*Test 3:* Applied to each key and keying material held in non-volatile memory and subject to destruction by overwrite by the TOE.
+Test 3 [conditional]: If the TOE the TOE supports directly examining the non-volatile memory, this test is applied to each key and keying material held in non-volatile memory and subject to destruction by overwrite by the TOE.
 
-. Record memory of the key or keying material.
+. Record the memory location of the key or keying material.
 . Cause the TOE to perform normal cryptographic processing with the key from Step #1.
 . Cause the TOE to destroy the key. Record the value to be used for the overwrite of the key. 
-. Examine the memory from Step #1. to ensure the appropriate pattern (recorded in Step #3) is used. 
+. Examine the memory location from Step #1. to ensure the appropriate pattern (recorded in Step #3) is used.
+
+The test succeeds if correct pattern is found at the memory location. If the pattern is not found, then the test fails.
+
+Test 4 [conditional]: If the TOE does not support directly examining the SDO/SDE memory or non-volatile memory, this test is applied to each key and keying material held in volatile memory or non-volatile memory.
+
+. Record the value or a corresponding checksum value of the key or keying material for a given key reference.
```

Comment:
Remove "the value or" because it is the conditional branch for TOE does not support directly examining memory and hence the key value cannot be recorded.

Reply:
I was thinking about a scenario where the evaluator cannot dump the (full) memory of the TOE, but the TOE itself still has some functionality to output key values. I'm not opposed to your suggestion though.